src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -925,4 +925,21 @@ public function testGetEnvCastsNull($expected, string $prefix)
             });
         }));
     }
+
+    /**
+     * @dataProvider provideGetEnvDefined
+     */
+    public function testGetEnvDefined(bool $expected, ?string $resolved)
+    {
+        $this->assertSame(
+            $expected,
+            (new EnvVarProcessor(new Container()))->getEnv('defined', '', fn () => $resolved),
+        );
+    }
+
+    public static function provideGetEnvDefined(): iterable
+    {
+        yield 'Defined' => [true, 'foo'];
+        yield 'Empty' => [false, ''];
```

Comment:
Nice try, but `empty('0') === true`. You should add this case: `[true, '0']`.

Reply:
Fixed!